### PR TITLE
Add general form list drag and drop reorder #535

### LINKED ISSUE #535: [Feature Request] provide function to change the order of defined contacts
**Is your feature request related to a problem? Please describe.**
Usually researcher are interested in defining a specific author order. Currently one can not change the order of the contacts without deleting exisiting ones, e.g. adding a contact in the middle means deleting the last contacts and adding them again. 

**Describe the solution you'd like**
Maybe some simple  "+" and "-" buttons to move the contacts in the list up and down

### PR DESCRIPTION
**Pull request by Freymaurer**
closes #535

form-reorder

### CODE REVIEW

**Comment by arendd** [comment]
Looks nice, functionality is great, only wondering about the icon with the "six dots" ? I'm not sure if this is obvious for the user that this if for drag&drop ? But only my personal impression.... THANK YOU

**Comment by Freymaurer** [comment]
Good feedback! Do you have another icon in mind? We can easily use all free tier icons from https://fontawesome.com/

File: src/Client/SharedComponents/Metadata/Forms.fs [hunk 1]
```diff
@@ -286,38 +286,126 @@ module private Helper =
...
 type FormComponents =
 
     [<ReactComponent>]
-    static member InputSequence<'A>(inputs: ResizeArray<'A>, constructor: unit -> 'A, setter: ResizeArray<'A> -> unit, inputComponent: 'A * ('A -> unit) * (MouseEvent -> unit) -> ReactElement, ?label: string) =
-        Bulma.field.div [
-            if label.IsSome then Bulma.label label.Value
-            Bulma.field.div [
-                Html.orderedList [
-                    prop.className "grid grid-cols-1 gap-2"
+    static member InputSequenceElement(key: string, id: string, listComponent: ReactElement) =
```
Comment (Etschbeijer) [inline src/Client/SharedComponents/Metadata/Forms.fs:295]: You have the parameter key that is never used in the method

Comment (Freymaurer) [reply src/Client/SharedComponents/Metadata/Forms.fs:295]: that is correct! it is not used in the method, but when transpiled to jsx this is similiar to

```tsx
<InputSequenceElement key={key} />
```

Which is the react required way keep react states on possible reorder events. [more](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key)

This looks strange but it is not necessary to use key inside the component, as simply passing the key parameter already sets key for react.

I actually only found out about this by looking into the Feliz.React source code. I am working on a PR to expand the Feliz docs accordingly

**Comment by arendd** [comment]
What about one of these:
https://fontawesome.com/icons/arrows-up-down-left-right?f=classic&s=solid
https://fontawesome.com/icons/arrows-up-down?f=classic&s=solid

**Comment by Freymaurer** [comment]
> What about one of these: https://fontawesome.com/icons/arrows-up-down-left-right?f=classic&s=solid https://fontawesome.com/icons/arrows-up-down?f=classic&s=solid

i updated to arrows-up-down 👍

**Review by Etschbeijer — APPROVED** [review]

**Pull request merged**